### PR TITLE
[MISC] Alphabet: rename phred68legacy to phred68solexa

### PR DESCRIPTION
**Pull request by Irallia**
Resolves: https://github.com/seqan/product_backlog/issues/315

### CODE REVIEW

**Comment by marehr** [comment]
Snippets failed

```
 [ 65%] Building CXX object CMakeFiles/phred68solexa_snippet.dir/alphabet/quality/phred68solexa.cpp.o
In file included from /home/runner/work/seqan3/seqan3/seqan3/test/snippet/alphabet/quality/phred68solexa.cpp:1:0:
/home/runner/work/seqan3/seqan3/seqan3/include/seqan3/alphabet/quality/phred68legacy.hpp:26:13: error: This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/alphabet/quality/phred68solexa.hpp> instead. [-Werror]
    "This header is deprecated and will be removed in SeqAn-3.1.0; Please #include <seqan3/alphabet/quality/phred68solexa.hpp> instead.")
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~        
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/phred68solexa_snippet.dir/build.make:63: CMakeFiles/phred68solexa_snippet.dir/alphabet/quality/phred68solexa.cpp.o] Error 1
make[2]: Target 'CMakeFiles/phred68solexa_snippet.dir/build' not remade because of errors.
make[1]: *** [CMakeFiles/Makefile2:10534: CMakeFiles/phred68solexa_snippet.dir/all] Error 2
```

**Review by MitraDarja — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 
 * Added `seqan3::phred94`, a quality type that represents the full Phred Score range (Sanger format) and is used for
   PacBio Phred scores of HiFi reads ([\#2290](https://github.com/seqan/seqan3/pull/2290)).
+* Deprecated `seqan3::phred68legacy`. Use `seqan3::phred68solexa` instead.
+  ([\#...](https://github.com/seqan/seqan3/pull/...)).
```
Comment (eseiler) [inline CHANGELOG.md:36]:
```suggestion
  ([\#2522](https://github.com/seqan/seqan3/pull/2522)).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 
 * Added `seqan3::phred94`, a quality type that represents the full Phred Score range (Sanger format) and is used for
   PacBio Phred scores of HiFi reads ([\#2290](https://github.com/seqan/seqan3/pull/2290)).
+* Deprecated `seqan3::phred68legacy`. Use `seqan3::phred68solexa` instead.
+  ([\#...](https://github.com/seqan/seqan3/pull/2522)).
```
Comment (eseiler) [inline CHANGELOG.md:36]:
```suggestion
  ([\#2522](https://github.com/seqan/seqan3/pull/2522)).
```

Comment (eseiler) [inline CHANGELOG.md:36]: Can you move this down to the API changes section?

File: CHANGELOG.md [hunk 3]
```diff
@@ -60,7 +62,7 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
   ([\#2340](https://github.com/seqan/seqan3/pull/2340), [\#2380](https://github.com/seqan/seqan3/pull/2380),
   [\#2389](https://github.com/seqan/seqan3/pull/2389))
 * Deprecated `seqan3::field::seq_qual`. Use `seqan3::fiel::seq` and `seqan3::field::qual` instead.
-  ([\#2379](https://github.com/seqan/seqan3/pull/2379)). Check out 
+  ([\#2379](https://github.com/seqan/seqan3/pull/2379)). Check out
   [SeqAn3 Cookbook - Write Record](https://docs.seqan.de/seqan/3.0.3/cookbook.html)
   for usage.
```
Comment (eseiler) [inline CHANGELOG.md:67]: Can you move this down to the API changes section (while we are at it)?

File: CHANGELOG.md (hunk 2 above)
Comment (eseiler) [reply CHANGELOG.md:36]: I would rather write `renamed X to Y`
Deprecation is usually more for stuff that does not exist anymore

**Review by eseiler — CHANGES_REQUESTED** [review]
💅  + the stuff I wrote as comment before

File: CHANGELOG.md [hunk 4]
```diff
@@ -148,6 +145,9 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
 
 #### I/O
 
+* Deprecated `seqan3::field::seq_qual`. Use `seqan3::fiel::seq` and `seqan3::field::qual` instead.
```
Comment (eseiler) [inline CHANGELOG.md:148]:
```suggestion
* Deprecated `seqan3::field::seq_qual`. Use `seqan3::field::seq` and `seqan3::field::qual` instead.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -128,9 +124,12 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
   ([\#2388](https://github.com/seqan/seqan3/pull/2388)).
 * The literal 'M'_cigar_op was renamed to 'M'_cigar_operation
   ([\#2388](https://github.com/seqan/seqan3/pull/2388)).
+
+* Renamed `seqan3::phred68legacy` to `seqan3::phred68solexa` ([\#2522](https://github.com/seqan/seqan3/pull/2522)).
 * Renamed `seqan3::sam_dna16` to `seqan3::dna16sam`
   ([\#2521](https://github.com/seqan/seqan3/pull/2521)).
 
+
```
Comment (eseiler) [inline CHANGELOG.md:132]:
```suggestion
* Renamed `seqan3::phred68legacy` to `seqan3::phred68solexa` ([\#2522](https://github.com/seqan/seqan3/pull/2522)).
* Renamed `seqan3::sam_dna16` to `seqan3::dna16sam`
  ([\#2521](https://github.com/seqan/seqan3/pull/2521)).

```

**Review by eseiler — APPROVED** [review]
Looks good, can you resolve the conflict?

**Pull request merged**